From patchwork Wed Feb 26 20:12:37 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407133
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 70F80930
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:07 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 4252F24672
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:07 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="tvBy16g+"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727335AbgBZUNG (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:06 -0500
Received: from mail-pl1-f202.google.com ([209.85.214.202]:37197 "EHLO
        mail-pl1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727240AbgBZUNG (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:06 -0500
Received: by mail-pl1-f202.google.com with SMTP id t12so350920plo.4
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:05 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=RIB7zoLlqFIQoJvqoI2dpWIxGsE/dMGJkqPhxqkMWkE=;
        b=tvBy16g+l1UdG6TNTnJ2Mq7UwRcD9jLdze/j7ZtM8llI83AtzFBSamQCQiyvV9tzXO
         RNGdcZbVTTD5IKQWO52jRwmcXSVzX1tdhQpRiN+MAkYkr7WdFdn9ysBj5Tj9YFhi1GRM
         CNkHJWqNE3o78dnk0P/PxJXWHkr0cHEK01ah7suwahUnC8y41z0XP5A6fcRgUI13L/Sw
         OZqwlI6GWJB2AfC0DjfSYLCBpls74+4bst5xR4fE219ixvo/PQM0my+FAEtRX6RFbefv
         qU/Lgr0Akf6MLzyziSboAWjEPWos/PNrjNxZtVV1o6OZ30eVHW28XzLaODyXYLDK9ClN
         qJLw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=RIB7zoLlqFIQoJvqoI2dpWIxGsE/dMGJkqPhxqkMWkE=;
        b=BjW81TptoQV4F0I/ccPLPdTWTlhVaKzqCSdPejCDtfQmRcz+8VpP3NFJofNRvcFgoH
         3nSMjyX9v0sy/HOCXukbPDjACR07LexdntXM4IE5jemK7qh/g57imHoSViH8oxQFWT6x
         sOkROGlMWzjmIoIWlnlghaepGuexzxrfadjhNw2S+x4WIhwKpmJw44h/lGOi/NEPYjPs
         mliKBieqgZ4otBcsTo4f8xh8YZZPSecI7fwlbAMca2MLIpDFqmZkfGvH24vIgDjcsN+M
         bfKoPx/XJnuH0h1/7155jGMQj8MzHUW/lMbpw2bWIA4/WpbYKSDfHQ2DO/0UhnJVJmvq
         MkYg==
X-Gm-Message-State: APjAAAXuRtLT72FT8pmPnhYo6kzNhfi/HohxONE6uRIsztizu6XcMs1J
        HAA3HfWZgEdHDk8HKjAyPvHLsGIIXZGVNGE1b7RwUirakS0BukiH5uG3f2FXJAPq4UD5l3Y4o2B
        COC4NI9UvdLFW3i3ZVjt2ZsbTDnE/VM7wnQ73Xi0siMpmUEqYXUihnQ==
X-Google-Smtp-Source: 
 APXvYqy1OUTt879OCKhac5X76HmflHSqbTYXTIuLsgQzglZqBNOdO7pUAZ1q+tSVPLhqmJBM8B/eEFxjtQ==
X-Received: by 2002:a63:6101:: with SMTP id v1mr527939pgb.318.1582747985132;
 Wed, 26 Feb 2020 12:13:05 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:37 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-2-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 1/7] x86: emulator: use "SSE2" for the
 target
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The movdqu and movapd instructions are SSE2 instructions. Clang
interprets the __attribute__((target("sse"))) as allowing SSE only
instructions. Using SSE2 instructions cause an error.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/emulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/emulator.c b/x86/emulator.c
index 8fe03b8..2990550 100644
--- a/x86/emulator.c
+++ b/x86/emulator.c
@@ -658,7 +658,7 @@ static bool sseeq(sse_union *v1, sse_union *v2)
     return ok;
 }
 
-static __attribute__((target("sse"))) void test_sse(sse_union *mem)
+static __attribute__((target("sse2"))) void test_sse(sse_union *mem)
 {
     sse_union v;
 

From patchwork Wed Feb 26 20:12:38 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407137
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id B1FA614B4
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:11 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 8717F24672
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:11 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="wAINI25m"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727357AbgBZUNK (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:10 -0500
Received: from mail-pf1-f201.google.com ([209.85.210.201]:53091 "EHLO
        mail-pf1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727240AbgBZUNK (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:10 -0500
Received: by mail-pf1-f201.google.com with SMTP id v26so197906pfn.19
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:08 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=WcjOhYarYrSlP0Y28DK/vHN2LbfwP4ntC96ZmKq5xCU=;
        b=wAINI25mJjJl67tX1Kffr6+OKjPnpiSQKExRYDmrxJsVK4yeiTII9ytqqzh0nNBfnm
         Z/PrSjhM+zvhzaE+E3T7RD9l2EzThsjdhZGMvaNgqp+J4bnXi91bByjJjZGGzBQiEpbM
         GGewkwAEJHuyaUGlL2rqkAdsFRNsHkeT8gklQ8xPkwybnGh+as4mf7MevI4uaCRxUXZy
         bb+h7bGyMiy4an1WLnoJ8doRV86KWxbTgga/sgJnLIJD/tdGiC3N8VIAt+kRQI2JX7EJ
         y5v8974mpnvmVCOSLBFRKFB0d4WbfzuyycTU+NA+bQ7Jkf0ntO7vZJvQNY1ryQ3fXduV
         ymgQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=WcjOhYarYrSlP0Y28DK/vHN2LbfwP4ntC96ZmKq5xCU=;
        b=DF6MHrVYPvqu2wx+aBlBDBsykKtXUlZD30qIYF15QrlhAUGPgNvN1J+opu/VydgxJI
         ln85BYRZClP9aQR+MfiMHGhPKuvm3pydYy3x1n1KwPwJ+KfXTheLOGe4TTAA0h30k9nY
         S1sJzyUtXIIzJUCQ/t7ooWGFACEffrrv//GtkOrJpRei4ZljpVm1fM9VxbABpWUP4OK5
         czNRok2QTfuGLCUZc+GB95NPapxfs4XhkZ64KA3xZ48dmhUBvdt3VHRMP11Ovwq9JOqy
         MZmFmk1ehsQ2yyJLxw9QqE0BrazaKfg1fB5wxy5tzA2I/zYs6tMW1cDJzxTbaLK0F3Ra
         H2wg==
X-Gm-Message-State: APjAAAXsntejhiVrUnuANgb5N2Q+xRc9cimG+Vhb8IDn2LtOeHsmijoC
        2/DBPcp9LnG+YfWa4h94DzmNRns5nE4RFWUCUbhCebHXvpi2p1/6aCtohmXI1HmX0rW7npEU+a8
        w++jtweEAlgXX4udgOVQsxq4RE5tapw/HiBh/09aYNP6VV59eC95y3Q==
X-Google-Smtp-Source: 
 APXvYqxjQAOoT667LkNmXZwdmTFvsX2Z9Zt9+bUyXEroptKsm3SvG9u6lzkDS3mrqM8V7mlzEKuqreY40Q==
X-Received: by 2002:a63:c546:: with SMTP id g6mr495020pgd.243.1582747987695;
 Wed, 26 Feb 2020 12:13:07 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:38 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-3-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 2/7] pci: cast masks to uint32_t for
 unsigned long values
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The "pci_bar_*" functions use 64-bit masks, but the results are assigned
to 32-bit variables. Cast mask usages to 32-bit, since we're interested
only in the least significant 4-bits.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 lib/pci.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/pci.c b/lib/pci.c
index daa33e1..1b85411 100644
--- a/lib/pci.c
+++ b/lib/pci.c
@@ -107,7 +107,8 @@ pcidevaddr_t pci_find_dev(uint16_t vendor_id, uint16_t device_id)
 uint32_t pci_bar_mask(uint32_t bar)
 {
 	return (bar & PCI_BASE_ADDRESS_SPACE_IO) ?
-		PCI_BASE_ADDRESS_IO_MASK : PCI_BASE_ADDRESS_MEM_MASK;
+		(uint32_t)PCI_BASE_ADDRESS_IO_MASK :
+		(uint32_t)PCI_BASE_ADDRESS_MEM_MASK;
 }
 
 uint32_t pci_bar_get(struct pci_dev *dev, int bar_num)

From patchwork Wed Feb 26 20:12:39 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407139
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 12910930
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:13 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id E3E8724670
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:12 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="TioLfC1M"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727387AbgBZUNM (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:12 -0500
Received: from mail-pj1-f73.google.com ([209.85.216.73]:54895 "EHLO
        mail-pj1-f73.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727240AbgBZUNL (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:11 -0500
Received: by mail-pj1-f73.google.com with SMTP id a31so137648pje.4
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:10 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=P0e2ipiL6n+VrTnPl5Byp+QGQep34dwDl9B8PERVjfw=;
        b=TioLfC1MhQGyEFcxspDYuSYLf+ArzkJHbrBRXHn7lXIRFjknUBtVosQ934e9h63WLF
         GEwRoDsu66F81+O4yvVsQLFZuOpYJladYzq8sVNHCY5wMqspnsrfsY44noYPgxAImjRT
         UBi59e1DIHB5PA7/0x23HEuvtWm78v3TFduLJfnqddBXPP8GmusxxH6F+t4NiS7RQevR
         jj39f5L6eYygcESz2voreMZDmeE/spW2f2h6Wh2ZznDvmbcYc7xCVNMyIy7Jy/A/cNVi
         gQFSJ1hn5vZR5df3HxOu5m3FfQ4nvEnVF/HWW3f/AaDbQcNBBLxyPd2ZJaaK2lCq6+RC
         mCtA==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=P0e2ipiL6n+VrTnPl5Byp+QGQep34dwDl9B8PERVjfw=;
        b=IIfjximF3Eu/D0CPEv9Q1+TkkQ11o7qXKYeqH9DREB6VJGZv121iSjD5ImNT78LhMV
         ykvZMsf6r43smg6I81d0cg+cQBt26B3z/zJdUdt6s+6gYPXDx1P8oMCg+XncDxDXAoYM
         SvpcdYyKM4DLMQAkTFIJu1ubL1stSKrpBbx9YTEjx2ONuLKEMg0NWXrDzoc4xscws6ek
         nT8bp2YZIgOFGOOAI3YAtqRb0DEf/C0iyzL5kn1DZSP2neS+GTonAebaJg0TXhEc/UlB
         q5LqWbxWrQbyhllxuQANt7Dxe6dK4ojKlssrta3yub+m9gJA3VUXF2+cJx4uwBWIn2a9
         uvbg==
X-Gm-Message-State: APjAAAUzTKXKWvyYD65oBVQ0PIG+rB07+4VmOo/tEklGRJ7X6a2vxLkK
        FD5+JcrcVxabSypQssibIG1roIw8qjryrZsoREt/Gj/o27iMIbOscrpXpr0kA4GjeOBYo6fMiPS
        5+r2esjOIWLWveIVAfEPCrwl0JLxZ652IwHJJ1172aQxtk+Xf5BkEJA==
X-Google-Smtp-Source: 
 APXvYqy1BLT3SljsAvplIDvETxsC26mvbI1joOdR0I51FFJfjPZFg+X9baD5/Q+DW3uw85t7DgY/9HKFEw==
X-Received: by 2002:a63:f10c:: with SMTP id f12mr541826pgi.386.1582747990141;
 Wed, 26 Feb 2020 12:13:10 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:39 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-4-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 3/7] x86: realmode: syscall: add explicit
 size suffix to ambiguous instructions
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Clang requires explicit size suffixes for potentially ambiguous
instructions:

x86/realmode.c:1647:2: error: ambiguous instructions require an explicit suffix (could be 'cmpb', 'cmpw', or 'cmpl')
        MK_INSN_PERF(perf_memory_load, "cmp $0, (%edi)");
        ^
x86/realmode.c:1591:10: note: expanded from macro 'MK_INSN_PERF'
                      "1:" insn "\n"                            \
                       ^
<inline asm>:8:3: note: instantiated into assembly here
1:cmp $0, (%edi)
  ^

The 'w' and 'l' suffixes generate code that's identical to the gcc
version without them.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/realmode.c | 6 +++---
 x86/syscall.c  | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/x86/realmode.c b/x86/realmode.c
index f5967ef..31f84d0 100644
--- a/x86/realmode.c
+++ b/x86/realmode.c
@@ -1644,7 +1644,7 @@ static void test_perf_memory_load(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_load, "cmp $0, (%edi)");
+	MK_INSN_PERF(perf_memory_load, "cmpw $0, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
@@ -1657,7 +1657,7 @@ static void test_perf_memory_store(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_store, "mov %ax, (%edi)");
+	MK_INSN_PERF(perf_memory_store, "movw %ax, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
@@ -1670,7 +1670,7 @@ static void test_perf_memory_rmw(void)
 {
 	u32 cyc, tmp;
 
-	MK_INSN_PERF(perf_memory_rmw, "add $1, (%edi)");
+	MK_INSN_PERF(perf_memory_rmw, "addw $1, (%edi)");
 
 	init_inregs(&(struct regs){ .edi = (u32)&tmp });
 
diff --git a/x86/syscall.c b/x86/syscall.c
index b4f5ac0..b7e29d6 100644
--- a/x86/syscall.c
+++ b/x86/syscall.c
@@ -38,7 +38,7 @@ static void handle_db(struct ex_regs *regs)
 
 /* expects desired ring 3 flags in rax */
 asm("syscall32_target:\n"
-    "   cmp $0, code_segment_upon_db(%rip)\n"
+    "   cmpl $0, code_segment_upon_db(%rip)\n"
     "   jne back_to_test\n"
     "   mov %eax,%r11d\n"
     "   sysretl\n");

From patchwork Wed Feb 26 20:12:40 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407141
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 806B6930
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:15 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 5FE8E24672
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:15 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="acW9weaZ"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727443AbgBZUNO (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:14 -0500
Received: from mail-pg1-f201.google.com ([209.85.215.201]:50429 "EHLO
        mail-pg1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727444AbgBZUNO (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:14 -0500
Received: by mail-pg1-f201.google.com with SMTP id 22so271659pgw.17
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:13 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=OdYlpv2ga298KF/gq7X6DdauhYrmwjzn6r6GXhMlTFY=;
        b=acW9weaZrIae64ShTa5DEXYIEHCSozEW83xlO15vq6HZz7lT896h58NTTRaLqWoC7s
         k1bMAAgi8WvA1wFKjId1/+Ll5xKd9WQkT+6y6HK9qnsgbhszi/ISgG1anfIw+XHwrx6P
         g8fmZ9reIo+9h5uRKerCgiySOvBnElWo4BdxajxOKjCiICGesrN0Zz905P9xZSyzUJBZ
         xWEjALDxGkib9/A2tuisAFDmIraTKHsbABUh2zZ+eFtVFY7fzETic0Gc4Yf5SIYrWtv1
         UaIMXdKXvD00EYoWuDMHKcF8sNu7JSTa7TW1HTZMLnUBJRA6GPBe+t0J7jvhFpcz00rS
         GsdQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=OdYlpv2ga298KF/gq7X6DdauhYrmwjzn6r6GXhMlTFY=;
        b=Pg+IOjc+7UBzgQAp2PmkfdbEkNPk1GtDsOm+55OBFMG627xFJF1uBs7Neis+Uly/5K
         IFYwMTCby6YVvqoinMt96YGIgDynkMAPLiWmY43uxMIKIdvJeZpdjnzZ/5+NvzlBmJxW
         tjCDYvD8nFviLuV1sdyQW2r8Afae+GxtlNfLTx0umKEXIpExC3UIAxWn6o1C34vt9+WW
         CUnaC2GLp6lzWB/rckuYfOKKoIktqtk6pEQ0FKddJaXmFIdpiW3gYbVLr8fGVXF3HMmG
         P/BJwkl8i3uL7QPcy6Qp1nITGtYzFe+Uq2isJOiAN9VXjJM8j6h34mU1LxwLrm7x034H
         WxZA==
X-Gm-Message-State: APjAAAWZU92JmnDHZWd3Y6FszgRvKXx82trETVgKm2FAdo1DPJ5EGEck
        qSMye9hf9tJdIMM8NteaINdNkQT8PrLgU6ZnIbH6NAb90XecpF0nr2w7HEKi01o61VjrBqUtefw
        2eF+T6gX00VI3wvRKOSG/8TcM3rrH4Adny4KCGC4LiuOWKz4TAGstbQ==
X-Google-Smtp-Source: 
 APXvYqzsj4/uYcHspUiV1VEsrYEp8PRW3TP7VSa28A4OliGP9du6wxTK7mxr1vR1v4V8gze6uohOVT9FKw==
X-Received: by 2002:a63:360a:: with SMTP id d10mr510305pga.366.1582747992896;
 Wed, 26 Feb 2020 12:13:12 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:40 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-5-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 4/7] svm: change operand to output-only for
 matching constraint
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

According to GNU extended asm documentation, "the two operands [of
matching constraints] must include one input-only operand and one
output-only operand." So remove the read/write modifier from the output
constraint.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/svm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/x86/svm.c b/x86/svm.c
index aa3d995..ae85194 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -288,8 +288,8 @@ static void test_run(struct test *test, struct vmcb *vmcb)
             "cli \n\t"
             "stgi"
             : // inputs clobbered by the guest:
-	      "+D" (the_test),            // first argument register
-	      "+b" (the_vmcb)             // callee save register!
+	      "=D" (the_test),            // first argument register
+	      "=b" (the_vmcb)             // callee save register!
             : [test] "0" (the_test),
 	      [vmcb_phys] "1"(the_vmcb),
 	      [PREPARE_GIF_CLEAR] "i" (offsetof(struct test, prepare_gif_clear))

From patchwork Wed Feb 26 20:12:41 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407143
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id A78E1930
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:17 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 85A6C21556
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:17 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="fFQpNZdU"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727460AbgBZUNQ (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:16 -0500
Received: from mail-pl1-f201.google.com ([209.85.214.201]:55191 "EHLO
        mail-pl1-f201.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727260AbgBZUNQ (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:16 -0500
Received: by mail-pl1-f201.google.com with SMTP id s13so320783plr.21
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:15 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=ToWeorAUFnUXwAhb7jaBBmGub+OaoJimWpyK5/Q9FSo=;
        b=fFQpNZdUYsBG0yoNSuVwfOheETmUiVXL4UgudFnyHagIFn53mTMuoz+P0bCQ+HGzwC
         8SMMMXZEJ2pBZeIj8SZEZ4O0Cj/kpZ2fVOqMChVU7Aj7Rpm6tZx4ypdA+RkKSdJCfcgX
         CyGO/EjYqdhlYZZ50LFD6URmB2Kzmvb/KTI6R4JpM4ygxBe9YRAUYI5q211ZPxrI1pU7
         yDHpZSeaCikg2GDVD3w9jlJWS3b2Kf705Oo7d+K29+9fcG2pe58a8GaOthb1dCboR+LE
         O4/l2poRzSScK6aADyu6QCKpFBMg/W+s9V+xGe9jncDzqpA1b6Mqv/uKc01zeXThCXKT
         ZsCg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=ToWeorAUFnUXwAhb7jaBBmGub+OaoJimWpyK5/Q9FSo=;
        b=dEut22CTAAc6miwDxEZvwBJUY5H/1aS2LiMshCc0VnPitQdgmmLDlgAejp7b8Cjeb5
         NEdrPTK/SFWPzhSDvWXT0Xu89vdnAznl1ja1Ojva3D3o3Ujc6wr3Sn7hWM+IuAFpMkhf
         zPaunvg0BFVnfM4mJmt+clxtTFiyOGOirQ4zZxEhzINK21K2e88qcP9EfsHKy3obciqw
         ZT7jlicnhU09TdO3dwe3pZy8tvsSf5Vq9GlRhssxOQc1q8RFZYIikmFiYRFMrZ7H3mAK
         Iz0yP8laeoF4DaUvdUYg4fikOcpCYc/LIwxHTx7vevdajET41kqNSFLbyhETyuJYtdJ1
         O0Zw==
X-Gm-Message-State: APjAAAULxh/h4XvsZCY9QFsgT64YqXERWJEgN/rkbtxCAepBkk5t2vGY
        nCCg65PmyShqQ2eJ5prymYdcylQGmbm/nM7rhiBaU//wpXfhmAKyFne1Nlx0IC4gz2upw/w1tbC
        pLagrpqnD4ajuLCV2ndhMrsxnffHnk0tSE/Eyj5t08KKi2IlhGk5Hrw==
X-Google-Smtp-Source: 
 APXvYqwl82W7+I6hrIYSq0LLhCkLCaPYYqDUcSxPSoUtm6xQ/iFxztxwS1tLpqxwsY7kxLldWi2TOib2jA==
X-Received: by 2002:a63:4e63:: with SMTP id o35mr480111pgl.279.1582747995137;
 Wed, 26 Feb 2020 12:13:15 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:41 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-6-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 5/7] svm: convert neg shift to unsigned
 shift
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Shifting a negative signed value is undefined. Use a shift of an
unsigned value instead.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/svm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86/svm.c b/x86/svm.c
index ae85194..17be4b0 100644
--- a/x86/svm.c
+++ b/x86/svm.c
@@ -1148,7 +1148,7 @@ static bool npt_rw_l1mmio_check(struct test *test)
 }
 
 #define TSC_ADJUST_VALUE    (1ll << 32)
-#define TSC_OFFSET_VALUE    (-1ll << 48)
+#define TSC_OFFSET_VALUE    (~0ull << 48)
 static bool ok;
 
 static void tsc_adjust_prepare(struct test *test)

From patchwork Wed Feb 26 20:12:42 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407145
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id C782014B4
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:19 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id A624B24670
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:19 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="oBMC1lt7"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727478AbgBZUNT (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:19 -0500
Received: from mail-pl1-f202.google.com ([209.85.214.202]:40546 "EHLO
        mail-pl1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727456AbgBZUNS (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:18 -0500
Received: by mail-pl1-f202.google.com with SMTP id y2so346987plt.7
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:18 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=uryJ2eJtKXOKe1NBEqWHoftgOZIJi7Y8qJbiVJC4L00=;
        b=oBMC1lt73FTYgaCNM3nuHuWGgV10AKArjOUyRD1q9cpxO7oQtUBVonZvTL+MB+tG5L
         Yc9kLSTX8pYR3Arjzudwdt7OJoYdNJwhM644mzY7lsYHhD2CrXAhASq9rrtVqMWrVb3y
         q5PpARK+Kx1TMas1/4Eykbk3YUIvI+nFozDfooRUbTbT7RGg5pcXc53ydCjHHaHT0RGp
         xnCT5AEdSqHEyJ8Pbaod6czbbSFyOvOOvjA+c0G9HDbVyjN1ay35ledJEQWdBnOa0tH1
         8ShU3IjwyQu5tfysCgFsNqPYrbNC6/R8qxSG6qkEd8qe70AtrjnS2XWl0QnT+YpPkOwG
         0GHQ==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=uryJ2eJtKXOKe1NBEqWHoftgOZIJi7Y8qJbiVJC4L00=;
        b=KD9mB0PTH1sjs7Ll9PftGMPjw2/9XVlSxwAW7UZ9ZgeA122FmLuxBRLxXz+4pcAXnj
         v80TvVEu2BlFjhaRiux0FtUnVjkgs5vfD/y6kR6MRGWgK/LBw5k/aD9aC11LPLEVo3Wy
         4OcOnoSyLb9IliJPDiHYfEcYW/qByZPoW1NqmzSKjLGofb+7lzb7NWJX77KCiQqrK8IT
         gNgcF0GznwvQUbFrJfy3Wtg5C05maCJIam80XD8X0bSflBflaNmjRW4LW1icmTW0UIy8
         OnXp8to5bUwfwFT8oXJ/zIR9B4pYIOOBoenXVPYGx9B97+ijuUFtwfCQegrUmyQ5pba4
         RiDw==
X-Gm-Message-State: APjAAAWUyX55c/v+5SrCxYxDxRzGHZ1nqPQphKpU7u6pUAr/7s+47mpq
        t0i++7LEy9O6zTKwB8WXWw5I/mfJCyHVhFIF5b3MYcK4+XxKskbd7poQb9gemwJFXL9isxjh8Mw
        cqO8vnKW/5h5pOSOmjlKtvk62hkyj0Yx22piRUjXkK7WcZb4haurk0Q==
X-Google-Smtp-Source: 
 APXvYqwtrTqGzOark/axu9TKwytDJxgzUcUNkQ1HaMrEzmRX0uy+OZ44KJbdei3O+BvJ0sn/f7t8v9af0A==
X-Received: by 2002:a63:fb04:: with SMTP id o4mr508775pgh.423.1582747997337;
 Wed, 26 Feb 2020 12:13:17 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:42 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-7-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 6/7] x86: VMX: use inline asm to get stack
 pointer
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

The only supported use of a local register variable is to specify
registers for input and output operands when calling Extended asm. Using
it to automatically collect the value in the register isn't supported as
the contents of the register aren't guaranteed. Instead use inline asm
to get the stack pointer explicitly.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/vmx_tests.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index a7abd63..ad8c002 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -2165,8 +2165,9 @@ static void into_guest_main(void)
 		.offset = (uintptr_t)&&into,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	uintptr_t rsp = 0;
 
+	asm volatile ("mov %%rsp, %0" : "=r" (rsp));
 	if (fp.offset != (uintptr_t)&&into) {
 		printf("Code address too high.\n");
 		return;
@@ -3261,8 +3262,9 @@ static void try_compat_invvpid(void *unused)
 		.offset = (uintptr_t)&&invvpid,
 		.selector = KERNEL_CS32,
 	};
-	register uintptr_t rsp asm("rsp");
+	register uintptr_t rsp = 0;
 
+	asm volatile ("mov %%rsp, %0" : "=r" (rsp));
 	TEST_ASSERT_MSG(fp.offset == (uintptr_t)&&invvpid,
 			"Code address too high.");
 	TEST_ASSERT_MSG(rsp == (u32)rsp, "Stack address too high.");

From patchwork Wed Feb 26 20:12:43 2020
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Bill Wendling <morbo@google.com>
X-Patchwork-Id: 11407147
Return-Path: <SRS0=q2E5=4O=vger.kernel.org=kvm-owner@kernel.org>
Received: from mail.kernel.org (pdx-korg-mail-1.web.codeaurora.org
 [172.30.200.123])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 3B97114B4
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:24 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.kernel.org (Postfix) with ESMTP id 0FB3024656
	for <patchwork-kvm@patchwork.kernel.org>;
 Wed, 26 Feb 2020 20:13:24 +0000 (UTC)
Authentication-Results: mail.kernel.org;
	dkim=pass (2048-bit key) header.d=google.com header.i=@google.com
 header.b="bEJ/WLER"
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1727471AbgBZUNX (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Wed, 26 Feb 2020 15:13:23 -0500
Received: from mail-pg1-f202.google.com ([209.85.215.202]:54830 "EHLO
        mail-pg1-f202.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1727456AbgBZUNW (ORCPT <rfc822;kvm@vger.kernel.org>);
        Wed, 26 Feb 2020 15:13:22 -0500
Received: by mail-pg1-f202.google.com with SMTP id l17so265644pgh.21
        for <kvm@vger.kernel.org>; Wed, 26 Feb 2020 12:13:20 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=google.com; s=20161025;
        h=date:in-reply-to:message-id:mime-version:references:subject:from:to
         :cc;
        bh=juAXt12km2i38jzPz/B2i0wdAoJlFZOk1KJ1Ao5kVQE=;
        b=bEJ/WLERUZo3ivhXzYuTnPOA9PVkOMnuKD2qz7Sgn6ajrBEj62ZmJFuslY081CSDPw
         klqjoshGAmxT7x8hvlfX/bXIX3SUlp1WZxUAXxLL274JYgi/CJB5k+FKzBdgA3fyIDJF
         r7PUjhz6UJECn2Sztnpik9WyE9fEhXLQrsCXmnKB+l8SyFftbCmXbAh2QDWJNYeezty6
         JuhOIMopnjd7cVXVrJgjTZMfHldomHpIcaJ/Um5ufin0RtYvjmryZBdF5ftOvSbuopsC
         D+RGZFPETVBjSDMrdKy7oGl2kRqVljTRkKpRgiWZVvuLjp5reKhh35kgBpVz81nUsBPp
         bdVw==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:date:in-reply-to:message-id:mime-version
         :references:subject:from:to:cc;
        bh=juAXt12km2i38jzPz/B2i0wdAoJlFZOk1KJ1Ao5kVQE=;
        b=Gw/oaNnBGCgSxd9GMsu20lpgxrr9mo71rVRKhTtrCA/jdmauyxLnPmnhzzSKLFPw1R
         IIlYCxUq5o5r48n/aELrBIAwRXfUKtfrYtgtgXKeELimvCS733njHmPTMWVTFPIXZ0g4
         hsc3uULMALs9V86RZRR7Q7JpIhZbX3CN/PyxNAfQa+OjuBidWTvsiVcVVJAka6L/JWww
         BEHxlmmhQ3PcggP/nomTkic6EzBi8wYYWnIg44px0gFqtRsw5uCrgO6I0wAHtLVWU8xw
         uAMdkSZR/MDp0E8j81x/taVOtSvCJWa/AJP/38Bue52mZei783yPHJJrYNciQfc/cGTB
         MA2w==
X-Gm-Message-State: APjAAAXcyOCnR3MHFNsNjqygUTvEPh/7HejlEENI5r7FKOmvD1NASn31
        C2U79xTxOFPCotCjyXfJjdXVEv3LBxyU1AeWVLIKix8B0Wy8k78nbUyGtrd6xOaXQcJ3uIB9I4p
        ieSWTGngaE3hyxk3NEvG6e86qyDH4sOSV6IExtrRh+zmkVi0EcVnk6Q==
X-Google-Smtp-Source: 
 APXvYqyFPMwO3UjdtV9KZS+Gmczt/6s35R5gwP1hWY2iDENsA0f8Wug5lB3xb2K4izUF2Wgc0QgUD3tjHg==
X-Received: by 2002:a63:d90c:: with SMTP id r12mr582562pgg.106.1582748000094;
 Wed, 26 Feb 2020 12:13:20 -0800 (PST)
Date: Wed, 26 Feb 2020 12:12:43 -0800
In-Reply-To: <20200226201243.86988-1-morbo@google.com>
Message-Id: <20200226201243.86988-8-morbo@google.com>
Mime-Version: 1.0
References: <20200226094433.210968-1-morbo@google.com>
 <20200226201243.86988-1-morbo@google.com>
X-Mailer: git-send-email 2.25.1.481.gfbce0eb801-goog
Subject: [kvm-unit-tests PATCH v3 7/7] x86: VMX: the "noclone" attribute is
 gcc-specific
From: Bill Wendling <morbo@google.com>
To: kvm@vger.kernel.org
Cc: pbonzini@redhat.com, oupton@google.com, drjones@redhat.com,
        Bill Wendling <morbo@google.com>
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org

Don't use the "noclone" attribute for clang as it's not supported.

Signed-off-by: Bill Wendling <morbo@google.com>
---
 x86/vmx_tests.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/x86/vmx_tests.c b/x86/vmx_tests.c
index ad8c002..ec88016 100644
--- a/x86/vmx_tests.c
+++ b/x86/vmx_tests.c
@@ -4976,7 +4976,10 @@ extern unsigned char test_mtf1;
 extern unsigned char test_mtf2;
 extern unsigned char test_mtf3;
 
-__attribute__((noclone)) static void test_mtf_guest(void)
+#ifndef __clang__
+__attribute__((noclone))
+#endif
+static void test_mtf_guest(void)
 {
 	asm ("vmcall;\n\t"
 	     "out %al, $0x80;\n\t"
